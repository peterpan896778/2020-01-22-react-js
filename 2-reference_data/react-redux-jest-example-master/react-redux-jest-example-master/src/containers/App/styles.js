import styled from 'styled-components';

export const Container = styled.div`
  max-width: 1440px;
  margin: auto;
  text-align: center;
  font-family: 'Open Sans', sans-serif;
`;

export const H2 = styled.h2`
  font-weight: 600;
`;
