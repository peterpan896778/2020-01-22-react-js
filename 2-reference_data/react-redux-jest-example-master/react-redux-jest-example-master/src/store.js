import { createStore } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import reducer from 'src/reducers';

const store = createStore(
  reducer,
  composeWithDevTools(),
);

export default store;
