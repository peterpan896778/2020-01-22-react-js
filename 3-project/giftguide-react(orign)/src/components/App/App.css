@font-face {
  font-family: GothamRndSSm-Bold;
  src: url("../../assets/fonts/GothamRndSSm-Bold.otf");
}
@font-face {
  font-family: GothamRndSSm-BoldItalic;
  src: url("../../assets/fonts/GothamRndSSm-BoldItalic.otf");
}
@font-face {
  font-family: GothamRndSSm-Book;
  src: url("../../assets/fonts/GothamRndSSm-Book.otf");
}
@font-face {
  font-family: GothamRndSSm-BookItalic;
  src: url("../../assets/fonts/GothamRndSSm-BookItalic.otf");
}
@font-face {
  font-family: GothamRndSSm-Light;
  src: url("../../assets/fonts/GothamRndSSm-Light.otf");
}
@font-face {
  font-family: GothamRndSSm-LightItalic;
  src: url("../../assets/fonts/GothamRndSSm-LightItalic.otf");
}
@font-face {
  font-family: GothamRndSSm-Medium;
  src: url("../../assets/fonts/GothamRndSSm-Medium.otf");
}
@font-face {
  font-family: GothamRndSSm-MediumItalic;
  src: url("../../assets/fonts/GothamRndSSm-MediumItalic.otf");
}

body {
  background-color: rgba(25, 34, 48, 0.05);
  min-height: 1100px;
  margin: 0;
  padding: 0;
  font-family: "GothamRndSSm-Medium";
  font-style: normal;
  font-weight: 400;
}
