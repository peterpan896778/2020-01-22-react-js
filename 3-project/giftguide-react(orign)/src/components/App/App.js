import React, { Component, lazy, Suspense } from "react";
import { Route, Redirect, Switch, BrowserRouter } from "react-router-dom";
import { connect } from "react-redux";
import { bindActionCreators } from "redux";
import Loadable from "react-loadable";

// import { history } from "./../../stores";
import * as routes from "../../constants/routes";
import "./App.css";

import PageFactory from "../Pages/PageFactory";

// import DashboardPage from "../Pages/DashboardPage";
import AboutPage from "../Pages/AboutPage";
import AccountPage from "../Pages/AccountPage";
import AccountSettingPage from "../Pages/AccountSettingPage";
import UpdateWishItemPage from "../Pages/UpdateWishItemPage";
import BlogPage from "../Pages/BlogPage";
import CalendarPage from "../Pages/CalendarPage";
import FeedbackPage from "../Pages/FeedbackPage";
import FriendsPage from "../Pages/FriendsPage";
import GiftGuidesPage from "../Pages/GiftGuidesPage";
import PrivacyPolicyPage from "../Pages/PrivacyPolicyPage";
import ProfilePage from "../Pages/ProfilePage";
import SearchPage from "../Pages/SearchPage";
import WishListPage from "../Pages/WishListPage";
import ContactPage from "../Pages/ContactPage";
import delay from "delay";
const loading = () => <div>Loading</div>;

// All layouts/containers
const DashboardPage = Loadable({
  loader: async () => {
    return import("../Pages/DashboardPage");
  },
  // render(loaded, props) {
  //   let Component = loaded.default;
  //   return <Component {...props} />;
  // },
  loading
});

class App extends Component {
  constructor(props) {
    super(props);
    console.log("======= app.js props =================");
    console.log(props);
  }

  componentWillMount() {
    console.log("======= this is willmount of app =========");
  }

  render() {
    const DashboardPageAugmented = () => (
      <PageFactory user={this.props.user}>
        <Suspense fallback={loading()}>
          <DashboardPage />
        </Suspense>
      </PageFactory>
    );
    const AboutPageAugmented = () => (
      <PageFactory user={this.props.user}>
        <AboutPage />
      </PageFactory>
    );
    const AccountPageAugmented = () => (
      <PageFactory user={this.props.user}>
        <AccountPage />
      </PageFactory>
    );
    const AccountSettingPageAugmented = () => (
      <PageFactory user={this.props.user}>
        <AccountSettingPage />
      </PageFactory>
    );
    const UpdateWishItemPageAugmented = () => (
      <PageFactory user={this.props.user}>
        <UpdateWishItemPage />
      </PageFactory>
    );
    const BlogPageAugmented = () => (
      <PageFactory user={this.props.user}>
        <BlogPage />
      </PageFactory>
    );
    const CalendarPageAugmented = () => (
      <PageFactory user={this.props.user}>
        <CalendarPage />
      </PageFactory>
    );
    const FeedbackPageAugmented = () => (
      <PageFactory user={this.props.user}>
        <FeedbackPage />
      </PageFactory>
    );
    const FriendsPageAugmented = () => (
      <PageFactory user={this.props.user}>
        <FriendsPage />
      </PageFactory>
    );
    const WishListPageAugmented = () => (
      <PageFactory user={this.props.user}>
        <WishListPage />
      </PageFactory>
    );
    const SearchPageAugmented = () => (
      <PageFactory user={this.props.user}>
        <SearchPage />
      </PageFactory>
    );
    const ProfilePageAugmented = () => (
      <PageFactory user={this.props.user}>
        <ProfilePage />
      </PageFactory>
    );
    const GiftGuidesPageAugmented = () => (
      <PageFactory user={this.props.user}>
        <GiftGuidesPage />
      </PageFactory>
    );
    const PrivacyPolicyPageAugmented = () => (
      <PageFactory user={this.props.user}>
        <PrivacyPolicyPage />
      </PageFactory>
    );
    const ContactPageAugmented = () => (
      <PageFactory user={this.props.user}>
        <ContactPage />
      </PageFactory>
    );
    // prettier-ignore
    return (
      <BrowserRouter>
        <React.Fragment>
          <div className="app">
            {
              // (status === 'lapsed' || status === 'inactive' || status === '7day_expire') && this.props.location !== routes.ACCOUNT_PAYMENTS &&
              // this.props.location !== routes.ACCOUNT_PROFILE && this.props.location !== routes.ACCOUNT_PASSWORD ? (
              //   <Redirect to={routes.ACCOUNT_PAYMENTS} />
              // ) : null
            }
            <Switch>
              <Redirect exact from="/" to={routes.DASHBOARD_PAGE} />
              <Route exact path={routes.DASHBOARD_PAGE} component={DashboardPageAugmented} />
            </Switch>
            <Route exact path={ routes.SEARCH_PAGE } component={ SearchPageAugmented } />
            <Route exact path={ routes.WISH_LIST_PAGE } component={ WishListPageAugmented } />
            <Route exact path={routes.ABOUT_PAGE} component={AboutPageAugmented} />
            <Route exact path={routes.ACCOUNT_PAGE} component={AccountPageAugmented} />
            <Route exact path={routes.ACCOUNT_SETTING_PAGE}	component={AccountSettingPageAugmented} />
            <Route exact path={routes.UPDATE_WISH_ITEM_PAGE}	component={UpdateWishItemPageAugmented} />
            <Route exact path={routes.BLOG_PAGE}	component={BlogPageAugmented} />
            <Route exact path={routes.CALENDAR_PAGE} component={CalendarPageAugmented} />
            <Route exact path={routes.FEEDBACK_PAGE}	component={FeedbackPageAugmented} />
            <Route exact path={routes.FRIENDS_PAGE} component={ FriendsPageAugmented } />
            <Route exact path={routes.PROFILE_PAGE}	component={ProfilePageAugmented} />
            <Route exact path={routes.GIFTGUIDES_PAGE}	component={GiftGuidesPageAugmented} />
            <Route exact path={routes.PRIVACY_POLICY_PAGE}	component={PrivacyPolicyPageAugmented} />
            <Route exact path={routes.CONTACT_PAGE}	component={ContactPageAugmented} />
          </div>
        </React.Fragment>
      </BrowserRouter>
    );
  }
}

const mapStateToProps = state => {
  return {
    user: state.userReducer.user,
    loading: state.userReducer.authState
  };
};

// const mapDispatchToProps = dispatch =>
//   bindActionCreators({ userRequest }, dispatch);
const mapDispatchToProps = dispatch => bindActionCreators({}, dispatch);

// export default App;
export default connect(mapStateToProps, mapDispatchToProps)(App);
