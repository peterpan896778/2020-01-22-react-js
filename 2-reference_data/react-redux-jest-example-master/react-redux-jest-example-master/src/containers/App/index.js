import React from 'react';
import PropTypes from 'prop-types';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';

import * as actions from 'src/actions';
import { Counter, Logo } from 'src/components';

import logo from 'src/assets/logo.svg';
import { Container, H2 } from './styles';

const App = ({ counter, increment, decrement }) => (
  <Container>
    <Logo src={logo} />
    <H2>React Redux Jest Example</H2>
    <Counter
      decrement={decrement}
      increment={increment}
      value={counter}
    />
  </Container>
);

App.defaultProps = {
  counter: 0,
  decrement: () => {},
  increment: () => {},
};

App.propTypes = {
  counter: PropTypes.number,
  decrement: PropTypes.func,
  increment: PropTypes.func,
};

export default connect(
  state => ({ counter: state.counter.current }),
  dispatch => ({ ...bindActionCreators(actions, dispatch) }),
)(App);
