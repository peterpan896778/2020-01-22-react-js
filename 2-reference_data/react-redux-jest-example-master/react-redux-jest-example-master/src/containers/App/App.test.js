import React from 'react';
import { mount, shallow } from 'enzyme';
import store from 'src/store';
import App from '.';

test('<App /> mounts without crashing', () => {
  const wrapper = mount(
    <App store={store} />,
  );
  expect(wrapper).toMatchSnapshot();
});

test('<App /> renders', () => {
  const wrapper = shallow(
    <App store={store} />,
  );
  expect(wrapper).toMatchSnapshot();
});
