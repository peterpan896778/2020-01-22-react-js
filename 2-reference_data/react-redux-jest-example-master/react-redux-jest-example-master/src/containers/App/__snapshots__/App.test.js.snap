// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> mounts without crashing 1`] = `
<Connect(App)
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <App
    counter={0}
    decrement={[Function]}
    increment={[Function]}
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": [Function],
      }
    }
  >
    <styled.div>
      <div
        className="sc-ifAKCX iGUmiv"
      >
        <Logo
          src="logo.svg"
        >
          <styled.img
            src="logo.svg"
          >
            <img
              className="sc-bxivhb gxVCyX"
              src="logo.svg"
            />
          </styled.img>
        </Logo>
        <styled.h2>
          <h2
            className="sc-EHOje gLWsgx"
          >
            React Redux Jest Example
          </h2>
        </styled.h2>
        <Counter
          decrement={[Function]}
          increment={[Function]}
          value={0}
        >
          <div>
            <styled.div>
              <div
                className="sc-bwzfXH fyTSsl"
              >
                <Button
                  label="Decrement"
                  onClick={[Function]}
                  type="danger"
                >
                  <styled.button
                    label="Decrement"
                    onClick={[Function]}
                    type="danger"
                  >
                    <button
                      className="sc-bdVaJa bVdEId"
                      label="Decrement"
                      onClick={[Function]}
                      type="danger"
                    >
                      Decrement
                    </button>
                  </styled.button>
                </Button>
                <Button
                  label="Increment"
                  onClick={[Function]}
                  type="success"
                >
                  <styled.button
                    label="Increment"
                    onClick={[Function]}
                    type="success"
                  >
                    <button
                      className="sc-bdVaJa hjTpqZ"
                      label="Increment"
                      onClick={[Function]}
                      type="success"
                    >
                      Increment
                    </button>
                  </styled.button>
                </Button>
              </div>
            </styled.div>
            <styled.div>
              <div
                className="sc-htpNat jMtkWU"
              >
                <span>
                  Counter: 
                  0
                </span>
              </div>
            </styled.div>
          </div>
        </Counter>
      </div>
    </styled.div>
  </App>
</Connect(App)>
`;

exports[`<App /> renders 1`] = `
<App
  counter={0}
  decrement={[Function]}
  increment={[Function]}
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "notify": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "unsubscribe": null,
    }
  }
/>
`;
